File: financial-manager-api/Entities/Requests/RegisterRequest.cs
using System.ComponentModel.DataAnnotations;

namespace financial_manager.Entities.Requests
{
    public class RegisterRequest
    {
        [Required]
        [MinLength(8)]
        public string FullName { get; set; } = null!;
        [Required]
        [EmailAddress]
        public string Email { get; set; } = null!;
        [Required]
        [MinLength(12)]
        public string Password { get; set; } = null!;
        [Required]
        [Compare("Password", ErrorMessage = "Passwords don't match")]
        public string PasswordConfirm { get; set; } = null!;
    }
}

File: financialManagerUI/src/app/entities/requests/register.request.ts
export class RegisterRequest {
  public fullName: string;
  public email: string;
  public password: string;
  public passwordConfirm: string;

  constructor(
    fullName: string,
    email: string,
    password: string,
    passwordConfirm: string
  ) {
    this.fullName = fullName;
    this.email = email;
    this.password = password;
    this.passwordConfirm = passwordConfirm;
  }
}
